core/src/conn.rs

```diff
@@ -64,6 +64,33 @@ impl MongoConnection {
             bson::from_document(cmd_res).map_err(Error::DatabaseVersionDeserialization)?;
         Ok(build_info.data_lake.version)
     }
+
+    /// cancels all queries for a given statement id
+    pub fn cancel_queries_for_statement(&self, statement_id: Bson) -> Result<bool> {
+        // use $currentOp and match the comment field to identify any queries issued by the current statement
+        let admin_db = self.client.database("admin");
+        let current_ops_pipeline = vec![
+            doc! {"$currentOp": {}},
+            doc! {"$match": {"command.comment": statement_id}},
+        ];
+        let mut cursor = admin_db
+            .aggregate(current_ops_pipeline, None)
+            .map_err(Error::QueryExecutionFailed)?;
```

Comment:
```suggestion
        // use $currentOp and match the comment field to identify any queries issued by the current statement
        let current_ops_pipeline = vec![
            doc! {"$currentOp": {}},
            doc! {"$match": {"command.comment": statement_id}},
        ];
        let admin_db = self.client.database("admin");
        let mut cursor = admin_db
            .aggregate(current_ops_pipeline, None)
            .map_err(Error::QueryExecutionFailed)?;
```
[optional] This is a way to make the comment closer to what it's describing and move `admin_db` to closer to its use.